### DISCUSSION TITLE
how all you guy config prettier in nvim?

### DISCUSSION THREAD

**Discussion by yushaku** [LazyVim/LazyVim#1286] [Q&A]
I config prettier like the document in lazyvim.org guild. but i notice that every time i open nvim, and format code, it will create a new prettied-server run in the background. and i quit nvim, it is still there do not be killed.

when i open several projects then i will have a pun of prettied-server like this:

![image](https://github.com/LazyVim/LazyVim/assets/72312124/221ff1c9-ab20-41d2-833c-89ba7dea3e09)

**Comment by dpetka2001** [comment]
Well, I don't know how your repos/projects are structured, but `prettierd` chooses the `root` directory based on the patterns stated [here](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/lua/null-ls/builtins/formatting/prettierd.lua). If you would like to change that you could try [spawn directory](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTIN_CONFIG.md#spawn-directory) or [conditional sources](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTIN_CONFIG.md#conditional-sources) to define the root directory null-ls attaches to and maybe that will give you less processes of prettierd.
